server/base/src/test/java/org/apache/accumulo/server/ServerContextTest.java

```diff
@@ -135,7 +135,8 @@ public void testCanRun() {
     // ensure this fails with older versions; the oldest supported version is hard-coded here
     // to ensure we don't unintentionally break upgrade support; changing this should be a conscious
     // decision and this check will ensure we don't overlook it
-    final int oldestSupported = 8;
+    // as of 3.0 we will only support upgrades from 2.1
+    final int oldestSupported = AccumuloDataVersion.ROOT_TABLET_META_CHANGES;
```

Comment:
I'm not sure this comment expresses anything that the next line doesn't already express more clearly. As we move forward, the version names in this comment will get stale over time and the comment will need to be constantly updated. I'm not sure it's any more helpful than having the explicit assignment line right after.

Reply:
The comment was pre-existing and I found it helpful - it stopped me in my tracks to make sure that removing the previous versions was what was intended.  Removing upgrade versions should be rare and having it explicitly stated here served as a good reminder / check.

Reply:
I don't mean the whole comment... most of the comment is fine... I just don't think it's useful to hard-code the version numbers into the comment, because that's not really adding anything that the rest of the comment doesn't already have... but by hard-coding these version names into the comment, it is making it more that we have to change when we move forward, in order to keep the comment current. I just think that's creating unnecessary work for ourselves, and in this case, I don't think it makes it any more clear than what was there before.

Reply:
Addressed in 70df59f3c7